app/src/main/java/com/github/se/polyfit/ui/screen/PostInfoScreen.kt

```diff
@@ -2,34 +2,53 @@ package com.github.se.polyfit.ui.screen
 
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
 import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material3.CircularProgressIndicator
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
 import androidx.core.content.ContextCompat
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.github.se.polyfit.R
 import com.github.se.polyfit.model.post.Post
 import com.github.se.polyfit.ui.components.postinfo.PostCard
+import com.github.se.polyfit.viewmodel.post.ViewPostViewModel
 
 @Composable
-fun PostInfoScreen(posts: List<Post>, index: Int = 0) {
+fun PostInfoScreen(
+    posts: List<Post> = listOf(),
```

Comment:
It could be better to remove the posts and only use the ViewPostViewModel.

Reply:
This is there for backwards compability to avoid breaking any previous implementation